x/mongo/driver/integration/scram_test.go

```diff
@@ -27,10 +27,12 @@ func TestSCRAM(t *testing.T) {
 		t.Skip("Skipping because authentication is required")
 	}
 
-	server, err := testutil.Topology(t).SelectServerLegacy(context.Background(), description.WriteSelector())
+	server, err := testutil.Topology(t).SelectServer(context.Background(), description.WriteSelector())
+	noerr(t, err)
+	serverConnection, err := server.Connection(context.Background())
```

Comment:
Same comment here about closing this connection.

Reply:
Done.